lib/logflare/buffers/memory_buffer.ex

```diff
@@ -1,66 +1,58 @@
 defmodule Logflare.Buffers.MemoryBuffer do
   @moduledoc """
   This is an implementation of an in-memory buffer, using `:queue`.any()
-  All operations are syncronous.
+  All operations are synchronous.
   """
-  alias Logflare.{Buffers.Buffer, Buffers.MemoryBuffer}
-  use GenServer
-  @behaviour Buffer
-
-  # GenServer state and init callbacks
-  defstruct queue: nil
 
-  def start_link(opts \\ []) do
-    GenServer.start_link(__MODULE__, [], opts)
-  end
+  @behaviour Logflare.Buffers.Buffer
 
-  @impl true
-  def init(_) do
-    {:ok, %MemoryBuffer{queue: :queue.new()}}
-  end
+  use GenServer
 
   # API
 
-  @impl Buffer
-  def add(identifier, payload) do
-    GenServer.call(identifier, {:add, [payload]})
-  end
-
-  @impl Buffer
+  @impl Logflare.Buffers.Buffer
   def add_many(identifier, payloads) do
     GenServer.call(identifier, {:add, payloads})
   end
 
-  @impl Buffer
-  def pop(identifier) do
-    GenServer.call(identifier, {:pop, 1})
-  end
-
-  @impl Buffer
+  @impl Logflare.Buffers.Buffer
   def pop_many(identifier, number) do
     GenServer.call(identifier, {:pop, number})
   end
 
-  @impl Buffer
+  @impl Logflare.Buffers.Buffer
   def clear(identifier) do
     GenServer.call(identifier, :clear)
   end
 
-  @impl Buffer
+  @impl Logflare.Buffers.Buffer
   def length(identifier) do
     GenServer.call(identifier, :length)
   end
 
+  # GenServer state and init callbacks
+  defstruct queue: nil, size: 0, proc_name: nil
+
+  def start_link(opts \\ []) do
+    proc_name = opts[:proc_name] || opts[:name]
+
+    GenServer.start_link(__MODULE__, %{proc_name: proc_name}, opts)
+  end
+
+  @impl GenServer
+  def init(%{proc_name: proc_name}) do
+    {:ok, %__MODULE__{queue: :queue.new(), proc_name: proc_name || inspect(self())}}
```

Comment:
can remove the inspect as the proc_name should always be an identifier.
```suggestion
    {:ok, %__MODULE__{queue: :queue.new(), proc_name: proc_name || self()}}
```

Reply:
My reason for `inspect/1` there was to use it in logs and telemetry events, but maybe having it as an PID and changing when needed isn't that bad.

Reply:
Also, I planned to reduce the amount of cases where we need *named processes* and instead work with just PIDs when possible.